From 0ed302b3d9c9982141f5718a903b3b0e67377efd Mon Sep 17 00:00:00 2001
From: Axel Karjalainen <axel@axka.fi>
Date: Sun, 3 Dec 2023 17:00:26 +0200
Subject: [PATCH] Add main function to main.py

This is done to be able to use main.py in setuptools' console_scripts
---
 main.py | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/main.py b/main.py
index 143dfb4..4682da3 100644
--- a/main.py
+++ b/main.py
@@ -18,6 +18,9 @@ from app.classes.models.management import HelpersManagement
 from app.classes.shared.import_helper import ImportHelpers
 from app.classes.shared.websocket_manager import WebSocketManager
 
+logger = logging.getLogger(__name__)
+Console.cyan(f"Logging set to: {logger.level}")
+
 console = Console()
 helper = Helpers()
 if helper.check_root():
@@ -89,8 +92,7 @@ def setup_logging(debug=True):
         Console.critical(f"Unable to read logging config from {logging_config_file}")
 
 
-# Our Main Starter
-if __name__ == "__main__":
+def main():
     parser = argparse.ArgumentParser("Crafty Controller - A Server Management System")
 
     parser.add_argument(
@@ -117,9 +119,6 @@ if __name__ == "__main__":
     if args.verbose:
         Console.level = "debug"
 
-    # setting up the logger object
-    logger = logging.getLogger(__name__)
-    Console.cyan(f"Logging set to: {logger.level}")
     peewee_logger = logging.getLogger("peewee")
     peewee_logger.setLevel(logging.INFO)
 
@@ -323,3 +322,6 @@ if __name__ == "__main__":
             time.sleep(1)
         tasks_manager._main_graceful_exit()
         crafty_prompt.universal_exit()
+
+if __name__ == "__main__":
+    main()
-- 
2.42.0

